examples/nlp/language_modeling/megatron_gpt_quantization.py

```diff
@@ -59,31 +65,45 @@ def get_calib_dataloader(data="cnn_dailymail", batch_size=64, calib_size=512, ma
         yield batch
 
 
-@hydra_runner(config_path="conf", config_name="megatron_quantization")
+@hydra_runner(config_path="conf", config_name="megatron_gpt_quantization")
 def main(cfg) -> None:
     if not torch.cuda.is_available():
-        raise EnvironmentError("GPU is required for the inference.")
+        raise EnvironmentError("GPU is required for the quantization.")
 
-    quantizer = Quantizer(cfg.quantization, cfg.inference, cfg.export, cfg.trainer)
+    # Overwrite model config with the one from the model checkpoint and apply quantization modifications
+    model_cfg = load_config(cfg.model.restore_from_path)
+    model_cfg.update(cfg.model)
+    model_cfg = Quantizer.modify_model_config(model_cfg)
+
+    trainer = Trainer(strategy=NLPDDPStrategy(), **cfg.trainer)
+    model = MegatronGPTModel.restore_from(
+        restore_path=cfg.model.restore_from_path, override_config_path=model_cfg, trainer=trainer
+    )
+    model.freeze()
+
+    quantizer = Quantizer(cfg.quantization, cfg.export)
 
     # Quantization algorithm can be set to None. This is useful for baseline precision
     # accuracy validation. In this case only weights export step will be performed:
     if cfg.quantization.algorithm is not None:
-        dataloader = get_calib_dataloader(
+        data_iter = get_calib_data_iter(
             cfg.quantization.calib_dataset,
             cfg.inference.batch_size,
             cfg.quantization.num_calib_size,
             cfg.inference.max_context_length,
         )
-        dataloader = [data for data in dataloader]
-    else:
-        dataloader = None
+        dataloader = [data for data in data_iter]
 
-    model = quantizer.quantize(
-        cfg.model_file, dataloader, cfg.tensor_model_parallel_size, cfg.pipeline_model_parallel_size
-    )
+        def forward_loop(model):
+            # NOTE: Alternatively you can also use `model.forward_bwd_step(data_iter, forward_only=True)`
+            # if your model is setup for training.
+            model.set_inference_config(OmegaConf.to_container(cfg.inference))
```

Comment:
This inference config can / cannot go into Quantizer - why / why not?

Reply:
Whether user needs an inference config depends on what forward loop they use. If they use predict_step then only they need inference config so makes sense to have that with forward loop configuration in the user code instead of quantizer

Reply:
Got it, makes sense then, thanks